tests/NHSD.BuyingCatalogue.Identity.Api.UnitTests/Certificates/CustomIdentityServerBuilderExtensionTests.cs

```diff
@@ -0,0 +1,92 @@
+            {
+                UseDeveloperCredentials = true,
+                CertificatePath = "",
+                CertificatePassword = ""
+            };
+
+            var loggerMock = Mock.Of<ILogger>();
+            var serviceCollectionMock = new Mock<IServiceCollection>();
+            var sutMock = Mock.Of<IIdentityServerBuilder>(i => i.Services == serviceCollectionMock.Object);
+
+            sutMock.AddCustomSigningCredential(settings, loggerMock);
+
+            serviceCollectionMock.Verify(
+                c => c.Add(It.Is<ServiceDescriptor>(s => s.ServiceType == typeof(ISigningCredentialStore))),
+                Times.Exactly(1));
+            serviceCollectionMock.Verify(
+                c => c.Add(It.Is<ServiceDescriptor>(s => s.ServiceType == typeof(IValidationKeysStore))),
+                Times.Exactly(1));
+        }
+
+        [Test]
+        public void UseInvalidPath_ThrowsCertificateSettingsException()
+        {
+            
+            var settings = new CertificateSettings
+            {
+                UseDeveloperCredentials = false,
+                CertificatePath = "bad path",
+                CertificatePassword = "NHSD"
+            };
+
+            var loggerMock = Mock.Of<ILogger>();
+            var serviceCollectionMock = new Mock<IServiceCollection>();
+            var sutMock = Mock.Of<IIdentityServerBuilder>(i => i.Services == serviceCollectionMock.Object);
+            Assert.Throws<CertificateSettingsException>(() => sutMock.AddCustomSigningCredential(settings, loggerMock));
+        }
+
+        [Test]
+        public void UseValidPath_Sets_Certificate()
+        {
+            //write certificate file from embedded resource to disk
+            using var stream = Assembly.GetExecutingAssembly()
+                .GetManifestResourceStream(this.GetType().Namespace + ".certificate.pfx").ThrowIfNull();
+            using (var outputStream = File.Create("certificate.pfx"))
+            {
+                stream.CopyTo(outputStream);
+                stream.Flush();
+                stream.Close();
+            }
+
+            var settings = new CertificateSettings
+            {
+                UseDeveloperCredentials = false, CertificatePath = "certificate.pfx", CertificatePassword = "NHSD"
+            };
+
+            var loggerMock = Mock.Of<ILogger>();
+            var serviceCollectionMock = new Mock<IServiceCollection>();
+            var sutMock = Mock.Of<IIdentityServerBuilder>(i => i.Services == serviceCollectionMock.Object);
+
+            sutMock.AddCustomSigningCredential(settings, loggerMock);
+
+            serviceCollectionMock.Verify(
+                c => c.Add(It.Is<ServiceDescriptor>(s => s.ServiceType == typeof(ISigningCredentialStore))),
+                Times.Exactly(1));
+            serviceCollectionMock.Verify(
+                c => c.Add(It.Is<ServiceDescriptor>(s => s.ServiceType == typeof(IValidationKeysStore))),
+                Times.Exactly(1));
```

Comment:
The assertions here are the same as in `UseDeveloperCredentials_Sets_DeveloperCredentials` so this test would still pass if `AddDeveloperSigningCredentials` was invoked instead of `AddSigningCredential`.